### PR TITLE
fix: checking notifications when ready

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- One potential issue is that, we have already rendered the list without checking whether the notification has been initialized. It runs async so there could be a race condition. And since none of the checklist's deps is related to the process, we never get the correct value.
- Adds a try-catch for when the initialization fails in rare cases. Probably we should log it similarly to auth as it is a crucial part of the process.
- Update the checklist to wait for initialization.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-174 #done

### CODE REVIEW

**Review by rebelchris — COMMENTED** [review]
Where you maybe able to reproduce locally? (by maybe making the load fail or blocking it?)

It would make things easier if we can test the validation.

**Review by nensidosari — COMMENTED** [review]
Looks good. Have nothing to add on top of what Chris mentioned.

**Comment by sshanzel** [comment]
> Where you maybe able to reproduce locally? (by maybe making the load fail or blocking it?)

I can throw an error to the try-catch, but then when you click the button to Subscribe it runs the function below, which behind the scenes will work as intended as it calls the native browser function. I can maybe do an early return but by doing that, I don't think would replicate the issues we are getting. As of today, I think no one has yet to fully reproduce the issue, but IMO this is one potential case where we are also not rendering the option anymore if it was not initialized properly. Thoughts?

Screenshot 2024-02-20 at 10 30 12 PM

File: packages/shared/src/contexts/NotificationsContext.tsx [hunk 1]
```diff
@@ -189,41 +189,46 @@ export const NotificationsContextProvider = ({
...
-      const isGranted = globalThis.Notification?.permission === 'granted';
-      const [id, subscribed, externalId] = await Promise.all([
-        globalThis.OneSignal?.getRegistrationId(),
-        OneSignalReact.getSubscription(),
-        OneSignalReact.getExternalUserId(),
-      ]);
-      const isValidSubscription = subscribed && isGranted;
-      setOneSignal(OneSignalReact);
+    } catch (err) {
+      // we should maybe log the error somewhere if in case there was an issue with the initialization
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/contexts/NotificationsContext.tsx:229]: Seems like it is already initiated in the webapp, do I need to add anything?

[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/contexts/NotificationsContext.tsx:229]: You'd have to test if it handles by itself, or if you need to also use that trackEvent inside here.
Can't recall from top of my head.

**Comment by rebelchris** [comment]
@sshanzel Did you see my last message on the discussion. https://github.com/dailydotdev/apps/pull/2659#discussion_r1497002208

**Review by rebelchris — APPROVED** [review]

**Pull request merged**